build/ci.go

```diff
@@ -158,6 +158,13 @@ func doInstall(cmdline []string) {
 		build.MustRun(goinstall)
 		return
 	}
+	// If we are cross compiling to ARMv5 ARMv6 or ARMv7, clean any prvious builds
+	if *arch == "arm" {
+		os.RemoveAll(filepath.Join(runtime.GOROOT(), "pkg", runtime.GOOS+"_arm"))
+		for _, path := range filepath.SplitList(build.GOPATH()) {
+			os.RemoveAll(filepath.Join(path, "pkg", runtime.GOOS+"_arm"))
+		}
+	}
```

Comment:
This hack would still be needed AFAIK because `go build` will put stdlib object files in `GOROOT`, irrelevant of where "client" packages are placed.